### ISSUE TITLE
error during mvn clean install

### ISSUE THREAD

**Issue by Alireza-cman** [thingsboard/thingsboard#389]
hi everyone, 
I wanna build and install the source code on my mac.  but I encounter some error that I cant fix it. 
my maven version is: 
`Maven home: /usr/local/apache-maven/apache-maven-3.5.2
Java version: 9.0.1, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk-9.0.1.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.13", arch: "x86_64", family: "Mac"
`
the error is in the **TEST** part of **dao** section.
please help me. I am working on it for one day :(

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.thingsboard.server.dao.SqlDaoServiceTestSuite
2017-11-01 11:16:22,426 [main] ERROR hsqldb.db.HSQLDB5F73BCEAFF.ENGINE - could not reopen database
org.hsqldb.HsqlException: file input/output error: /tmp/testDb.log
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.error.Error.error(Unknown Source)
	at org.hsqldb.persist.Log.openLog(Unknown Source)
	at org.hsqldb.persist.Log.open(Unknown Source)
	at org.hsqldb.persist.Logger.open(Unknown Source)
	at org.hsqldb.Database.reopen(Unknown Source)
	at org.hsqldb.Database.open(Unknown Source)
	at org.hsqldb.DatabaseManager.getDatabase(Unknown Source)
	at org.hsqldb.DatabaseManager.newSession(Unknown Source)
	at org.hsqldb.jdbc.JDBCConnection.<init>(Unknown Source)
	at org.hsqldb.jdbc.JDBCDriver.getConnection(Unknown Source)
	at org.hsqldb.jdbc.JDBCDriver.connect(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:678)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:229)
	at org.thingsboard.server.dao.CustomSqlUnit.cleanUpDb(CustomSqlUnit.java:92)
	at org.thingsboard.server.dao.CustomSqlUnit.before(CustomSqlUnit.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.extensions.cpsuite.ClasspathSuite.run(ClasspathSuite.java:196)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.619 sec <<< FAILURE! - in org.thingsboard.server.dao.SqlDaoServiceTestSuite
org.thingsboard.server.dao.SqlDaoServiceTestSuite  Time elapsed: 1.618 sec  <<< ERROR!
java.lang.RuntimeException: Unable to clean up embedded hsqldb. Reason: file input/output error: /tmp/testDb.log
Caused by: java.sql.SQLException: file input/output error: /tmp/testDb.log
Caused by: org.hsqldb.HsqlException: file input/output error: /tmp/testDb.log

Running org.thingsboard.server.dao.JpaDaoTestSuite
Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.077 sec - in org.thingsboard.server.dao.JpaDaoTestSuite
Running org.thingsboard.server.dao.NoSqlDaoServiceTestSuite
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.306 sec <<< FAILURE! - in org.thingsboard.server.dao.NoSqlDaoServiceTestSuite
org.thingsboard.server.dao.NoSqlDaoServiceTestSuite  Time elapsed: 0.306 sec  <<< ERROR!
java.lang.ExceptionInInitializerError
Caused by: org.apache.cassandra.exceptions.ConfigurationException: Expecting URI in variable: [cassandra.config]. Found[cassandra.yaml]. Please prefix the file with [file:///] for local files and [file://<server>/] for remote files. If you are executing this from an external tool, it needs to set Config.setClientMode(true) to avoid loading configuration.

Results :

Tests in error: 
  NoSqlDaoServiceTestSuite.org.thingsboard.server.dao.NoSqlDaoServiceTestSuite » ExceptionInInitializer
  SqlDaoServiceTestSuite.org.thingsboard.server.dao.SqlDaoServiceTestSuite » Runtime

Tests run: 2, Failures: 0, Errors: 2, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Thingsboard ........................................ SUCCESS [  1.106 s]
[INFO] Thingsboard Server Commons ......................... SUCCESS [  0.072 s]
[INFO] Thingsboard Server Common Data ..................... SUCCESS [ 10.369 s]
[INFO] Thingsboard Server Common Messages ................. SUCCESS [  0.802 s]
[INFO] Thingsboard Server Common Transport components ..... SUCCESS [  0.383 s]
[INFO] Thingsboard Server DAO Layer ....................... FAILURE [ 11.710 s]
[INFO] Thingsboard Server Extensions API .................. SKIPPED
[INFO] Thingsboard Server Core Extensions ................. SKIPPED
[INFO] Thingsboard Extensions ............................. SKIPPED
[INFO] Thingsboard Server RabbitMQ Extension .............. SKIPPED
[INFO] Thingsboard Server REST API Call Extension ......... SKIPPED
[INFO] Thingsboard Server Kafka Extension ................. SKIPPED
[INFO] Thingsboard Server MQTT Extension .................. SKIPPED
[INFO] Thingsboard Server Transport Modules ............... SKIPPED
[INFO] Thingsboard HTTP Transport ......................... SKIPPED
[INFO] Thingsboard COAP Transport ......................... SKIPPED
[INFO] Thingsboard MQTT Transport ......................... SKIPPED
[INFO] Thingsboard Server UI .............................. SKIPPED
[INFO] Thingsboard Server Tools ........................... SKIPPED
[INFO] ThingsBoard Server Application ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 25.148 s
[INFO] Finished at: 2017-11-01T11:16:23+03:30
[INFO] Final Memory: 64M/212M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project dao: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/alireza/thingsboard/dao/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by ashvayka** [comment]
we do not support mac as a dev platform, but it should work fine. I think the issue is not related to our code, but to a library

**Comment by ssmaurya** [comment]
@ashvakya One of our development machines is Mac and it works for us. We used Cassandra on it. It did take time to setup. 
@Alireza-cman You can try mvn clean install -DskipTests and if it builds successfully then you try debugging database part.

**Comment by Alireza-cman** [comment]
@ssmaurya many thanks for your consideration. 
I have tested the ` mvn clean install -DskipTests` and it works for me. 
the problem is not about the Mac OS I think, cause I have tested the source files in ubuntu 16.04 and encountered the same problem. 
there is another problem that I can't handle it :) 
there are lots of error when I import the project in the intellj. most of error are related to setter and getter. I things this problem is generated of annotation that Intellij does not support it. 
I have tried my best but, I couldn't handle it.
